Addon development: creating a type with "gui" in its name causes a weird error.

### Skript/Server Version

```markdown
Skript 2.6.4
Server version 1.20.1
```


### Bug Description

Hello, I am working on updating my Addon Skript-AnvilGUI.

Whenever I try to rename the type codename to "anvilgui" or "virtualanvilgui", it prints an error in the console, and refers to the type as "virtualanvilguu"


### Expected Behavior

It should find the class "virtualanvilgui"

### Steps to Reproduce

Add the word "anvilgui" or "gui" to any classinfo in your addon.

### Errors or Screenshots

```
: #!#! 
[15:28:54 ERROR]: #!#! [Skript] Severe Error:
[15:28:54 ERROR]: #!#! Could not load testanvil.sk
[15:28:54 ERROR]: #!#! 
[15:28:54 ERROR]: #!#! Something went horribly wrong with Skript.
[15:28:54 ERROR]: #!#! This issue is NOT your fault! You probably can't fix it yourself, either.
[15:28:54 ERROR]: #!#! It looks like you are using some plugin(s) that alter how Skript works (addons).
[15:28:54 ERROR]: #!#! Here is full list of them:
[15:28:54 ERROR]: #!#! Skript-AnvilGUI v1.2 
[15:28:54 ERROR]: #!#! We could not identify which of those are specially related, so this might also be Skript issue.
[15:28:54 ERROR]: #!#! You should try disabling those plugins one by one, trying to find which one causes it.
[15:28:54 ERROR]: #!#! If the error doesn't disappear even after disabling all listed plugins, it is probably Skript issue.
[15:28:54 ERROR]: #!#! In that case, you will be given instruction on how should you report it.
[15:28:54 ERROR]: #!#! On the other hand, if the error disappears when disabling some plugin, report it to author of that plugin.
[15:28:54 ERROR]: #!#! Only if the author tells you to do so, report it to Skript's issue tracker.
[15:28:54 ERROR]: #!#! 
[15:28:54 ERROR]: #!#! Stack trace:
[15:28:54 ERROR]: #!#! java.lang.RuntimeException: pattern compiling exception, element class: org.derewah.skriptanvilgui.effects.EffOpenAnvilGUI
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:226)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:177)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.Statement.parse(Statement.java:60)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadItems(ScriptLoader.java:1114)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.command.Commands.loadCommand(Commands.java:500)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadScript(ScriptLoader.java:708)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.lambda$loadScripts$4(ScriptLoader.java:526)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.lambda$makeFuture$2(ScriptLoader.java:418)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.makeFuture(ScriptLoader.java:434)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:525)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:469)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.Skript$1.run(Skript.java:786)
[15:28:54 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftTask.run(CraftTask.java:101)
[15:28:54 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:480)
[15:28:54 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1112)
[15:28:54 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318)
[15:28:54 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:833)
[15:28:54 ERROR]: #!#! Caused by: ch.njol.skript.patterns.MalformedPatternException: caught exception while compiling pattern [pattern: open (anvilgui|anvil gui) %virtualanvilgui% to %player%]
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.patterns.PatternCompiler.compile(PatternCompiler.java:54)
[15:28:54 ERROR]: #!#!     at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1220)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1254)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:224)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:177)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.Statement.parse(Statement.java:60)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadItems(ScriptLoader.java:1114)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.command.Commands.loadCommand(Commands.java:500)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadScript(ScriptLoader.java:708)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.lambda$loadScripts$4(ScriptLoader.java:526)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.lambda$makeFuture$2(ScriptLoader.java:418)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.makeFuture(ScriptLoader.java:434)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:525)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:469)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.Skript$1.run(Skript.java:786)
[15:28:54 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftTask.run(CraftTask.java:101)
[15:28:54 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:480)
[15:28:54 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1112)
[15:28:54 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318)
[15:28:54 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:833)
[15:28:54 ERROR]: #!#! Caused by: ch.njol.skript.SkriptAPIException: No class info found for virtualanvilguus
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.registrations.Classes.getClassInfo(Classes.java:264)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.patterns.TypePatternElement.fromString(TypePatternElement.java:92)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.patterns.PatternCompiler.compile(PatternCompiler.java:138)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.patterns.PatternCompiler.compile(PatternCompiler.java:49)
[15:28:54 ERROR]: #!#!     at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1220)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.SkriptParser.parse_i(SkriptParser.java:1254)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:224)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:177)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.lang.Statement.parse(Statement.java:60)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadItems(ScriptLoader.java:1114)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.command.Commands.loadCommand(Commands.java:500)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadScript(ScriptLoader.java:708)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.lambda$loadScripts$4(ScriptLoader.java:526)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.lambda$makeFuture$2(ScriptLoader.java:418)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.makeFuture(ScriptLoader.java:434)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:525)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.ScriptLoader.loadScripts(ScriptLoader.java:469)
[15:28:54 ERROR]: #!#!     at Skript-2.6.4.jar//ch.njol.skript.Skript$1.run(Skript.java:786)
[15:28:54 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftTask.run(CraftTask.java:101)
[15:28:54 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:480)
[15:28:54 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1112)
[15:28:54 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318)
[15:28:54 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:833)
[15:28:54 ERROR]: #!#! 
[15:28:54 ERROR]: #!#! Version Information:
[15:28:54 ERROR]: #!#!   Skript: 2.6.4 (latest)
[15:28:54 ERROR]: #!#!     Flavor: skriptlang-github
[15:28:54 ERROR]: #!#!     Date: 17:14:25.041218900
[15:28:54 ERROR]: #!#!   Bukkit: 1.20.1-R0.1-SNAPSHOT
[15:28:54 ERROR]: #!#!   Minecraft: 1.20.1
[15:28:54 ERROR]: #!#!   Java: 17.0.7 (Java HotSpot(TM) 64-Bit Server VM 17.0.7+8-LTS-224)
[15:28:54 ERROR]: #!#!   OS: Windows 11 amd64 10.0
[15:28:54 ERROR]: #!#! 
[15:28:54 ERROR]: #!#! Server platform: Paper
[15:28:54 ERROR]: #!#! 
[15:28:54 ERROR]: #!#! Current node: open anvil gui {_x} to player (testanvil.sk, line 4)
[15:28:54 ERROR]: #!#! Current item: null
[15:28:54 ERROR]: #!#! 
[15:28:54 ERROR]: #!#! Thread: Server thread
[15:28:54 ERROR]: #!#! 
[15:28:54 ERROR]: #!#! Language: english
[15:28:54 ERROR]: #!#! Link parse mode: DISABLED
[15:28:54 ERROR]: #!#! 
[15:28:54 ERROR]: #!#! End of Error.
[15:28:54 ERROR]: #!#! 
```

### Other

_No response_

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

I came across a similar issue when trying to pluralize "objective" -> "objectives" in my addon, getting a similar error:
`Caused by: ch.njol.skript.SkriptAPIException: No class info found for objectife`

I have a lang file all setup correctly, and still a no go.

> I came across a similar issue when trying to pluralize "objective" -> "objectives" in my addon, getting a similar error: `Caused by: ch.njol.skript.SkriptAPIException: No class info found for objectife`
> 
> I have a lang file all setup correctly, and still a no go.

Error says `objectife` with an F. Did you mistype `objective`?

--- a/src/main/java/ch/njol/skript/util/Utils.java
+++ b/src/main/java/ch/njol/skript/util/Utils.java
@@ -70,9 +70,6 @@ public abstract class Utils {
 	protected final static Deque<WordEnding> plurals = new LinkedList<>();
 
 	static {
-		plurals.add(new WordEnding("ive", "ives")); // objective fix
-		plurals.add(new WordEnding("fe", "ves"));// most -f words' plurals can end in -fs as well as -ves
-
 		plurals.add(new WordEnding("axe", "axes"));
 		plurals.add(new WordEnding("x", "xes"));
 
@@ -84,6 +81,11 @@ public abstract class Utils {
 		plurals.add(new WordEnding("kie", "kies"));
 		plurals.add(new WordEnding("zombie", "zombies"));
 		plurals.add(new WordEnding("y", "ies"));
+
+		plurals.add(new WordEnding("wife", "wives")); // we have to do the -ife -> ives first
+		plurals.add(new WordEnding("life", "lives"));
+		plurals.add(new WordEnding("ive", "ives"));
+		plurals.add(new WordEnding("fe", "ves"));// most -f words' plurals can end in -fs as well as -ves
 
 		plurals.add(new WordEnding("h", "hes"));
 

--- a/src/main/java/ch/njol/skript/util/Utils.java
+++ b/src/main/java/ch/njol/skript/util/Utils.java
@@ -283,7 +283,7 @@ public abstract class Utils {
 	 * @return Pair of singular string + boolean whether it was plural
 	 */
 	@SuppressWarnings("null")
-	public static NonNullPair<String, Boolean> getEnglishPlural(final String word) {
+	public static NonNullPair<String, Boolean> getEnglishPlural(String word) {
 		assert word != null;
 		if (word.isEmpty())
 			return new NonNullPair<>("", Boolean.FALSE);
@@ -302,9 +302,9 @@ public abstract class Utils {
 	 * @param word
 	 * @return The english plural of the given word
 	 */
-	public static String toEnglishPlural(final String word) {
+	public static String toEnglishPlural(String word) {
 		assert word != null && word.length() != 0;
-		for (final WordEnding ending : plurals) {
+		for (WordEnding ending : plurals) {
 			if (word.endsWith(ending.singular()))
 				return word.substring(0, word.length() - ending.singular().length()) + ending.plural();
 		}

--- a/src/main/java/ch/njol/skript/util/Utils.java
+++ b/src/main/java/ch/njol/skript/util/Utils.java
@@ -94,6 +94,7 @@ public abstract class Utils {
 		plurals.add(new WordEnding("man", "men"));
 
 		plurals.add(new WordEnding("ui", "uis")); // gui fix
+		plurals.add(new WordEnding("api", "apis")); // api fix
 		plurals.add(new WordEnding("us", "i"));
 
 		plurals.add(new WordEnding("hoe", "hoes"));

--- a/src/main/java/ch/njol/skript/util/Utils.java
+++ b/src/main/java/ch/njol/skript/util/Utils.java
@@ -282,7 +282,6 @@ public abstract class Utils {
 	 * @param word trimmed string
 	 * @return Pair of singular string + boolean whether it was plural
 	 */
-	@SuppressWarnings("null")
 	public static NonNullPair<String, Boolean> getEnglishPlural(String word) {
 		assert word != null;
 		if (word.isEmpty())

--- a/src/main/java/ch/njol/skript/util/Utils.java
+++ b/src/main/java/ch/njol/skript/util/Utils.java
@@ -84,7 +84,9 @@ public abstract class Utils {
 
 		plurals.add(new WordEnding("wife", "wives")); // we have to do the -ife -> ives first
 		plurals.add(new WordEnding("life", "lives"));
+		plurals.add(new WordEnding("knife", "knives"));
 		plurals.add(new WordEnding("ive", "ives"));
+		plurals.add(new WordEnding("elf", "elves")); // self shelf elf
 		plurals.add(new WordEnding("fe", "ves"));// most -f words' plurals can end in -fs as well as -ves
 
 		plurals.add(new WordEnding("h", "hes"));
@@ -822,6 +824,19 @@ public abstract class Utils {
 
 		public String plural() {
 			return plural;
+		}
+
+		@Override
+		public boolean equals(Object object) {
+			if (this == object) return true;
+			if (!(object instanceof WordEnding)) return false;
+			WordEnding ending = (WordEnding) object;
+			return Objects.equals(singular, ending.singular) && Objects.equals(plural, ending.plural);
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(singular, plural);
 		}
 
 	}

--- a/src/main/java/ch/njol/skript/util/Utils.java
+++ b/src/main/java/ch/njol/skript/util/Utils.java
@@ -22,12 +22,7 @@ import java.io.File;
 import java.io.IOException;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Random;
+import java.util.*;
 import java.util.concurrent.CompletableFuture;
 import java.util.function.Predicate;
 import java.util.jar.JarEntry;
@@ -42,7 +37,6 @@ import org.bukkit.plugin.Plugin;
 import org.bukkit.plugin.java.JavaPlugin;
 import org.bukkit.plugin.messaging.Messenger;
 import org.bukkit.plugin.messaging.PluginMessageListener;
-import org.eclipse.jdt.annotation.Nullable;
 
 import com.google.common.collect.Iterables;
 import com.google.common.io.ByteArrayDataInput;
@@ -62,6 +56,7 @@ import ch.njol.util.StringUtils;
 import ch.njol.util.coll.CollectionUtils;
 import ch.njol.util.coll.iterator.EnumerationIterable;
 import net.md_5.bungee.api.ChatColor;
+import org.jetbrains.annotations.Nullable;
 import org.jetbrains.annotations.NotNull;
 
 /**
@@ -71,9 +66,48 @@ import org.jetbrains.annotations.NotNull;
  */
 public abstract class Utils {
 
-	private Utils() {}
-
 	public final static Random random = new Random();
+	protected final static Deque<WordEnding> plurals = new LinkedList<>();
+
+	static {
+		plurals.add(new WordEnding("ive", "ives")); // objective fix
+		plurals.add(new WordEnding("fe", "ves"));// most -f words' plurals can end in -fs as well as -ves
+
+		plurals.add(new WordEnding("axe", "axes"));
+		plurals.add(new WordEnding("x", "xes"));
+
+		plurals.add(new WordEnding("ay", "ays"));
+		plurals.add(new WordEnding("ey", "eys"));
+		plurals.add(new WordEnding("iy", "iys"));
+		plurals.add(new WordEnding("oy", "oys"));
+		plurals.add(new WordEnding("uy", "uys"));
+		plurals.add(new WordEnding("kie", "kies"));
+		plurals.add(new WordEnding("zombie", "zombies"));
+		plurals.add(new WordEnding("y", "ies"));
+
+		plurals.add(new WordEnding("h", "hes"));
+
+		plurals.add(new WordEnding("man", "men"));
+
+		plurals.add(new WordEnding("ui", "uis")); // gui fix
+		plurals.add(new WordEnding("us", "i"));
+
+		plurals.add(new WordEnding("hoe", "hoes"));
+		plurals.add(new WordEnding("toe", "toes"));
+		plurals.add(new WordEnding("o", "oes"));
+
+		plurals.add(new WordEnding("alias", "aliases"));
+		plurals.add(new WordEnding("gas", "gases"));
+
+		plurals.add(new WordEnding("child", "children"));
+
+		plurals.add(new WordEnding("sheep", "sheep"));
+
+		// general ending
+		plurals.add(new WordEnding("", "s"));
+	}
+
+	private Utils() {}
 
 	public static String join(final Object[] objects) {
 		assert objects != null;
@@ -239,77 +273,38 @@ public abstract class Utils {
 		return null;
 	}
 
-	private final static String[][] plurals = {
-
-			{"ive", "ives"}, // objective fix
-			{"fe", "ves"},// most -f words' plurals can end in -fs as well as -ves
-
-			{"axe", "axes"},
-			{"x", "xes"},
-
-			{"ay", "ays"},
-			{"ey", "eys"},
-			{"iy", "iys"},
-			{"oy", "oys"},
-			{"uy", "uys"},
-			{"kie", "kies"},
-			{"zombie", "zombies"},
-			{"y", "ies"},
-
-			{"h", "hes"},
-
-			{"man", "men"},
-
-		    {"ui", "uis"}, // gui fix
-			{"us", "i"},
-
-			{"hoe", "hoes"},
-			{"toe", "toes"},
-			{"o", "oes"},
-
-			{"alias", "aliases"},
-			{"gas", "gases"},
-
-			{"child", "children"},
-
-			{"sheep", "sheep"},
-
-			// general ending
-			{"", "s"},
-	};
-
 	/**
-	 * @param s trimmed string
+	 * @param word trimmed string
 	 * @return Pair of singular string + boolean whether it was plural
 	 */
 	@SuppressWarnings("null")
-	public static NonNullPair<String, Boolean> getEnglishPlural(final String s) {
-		assert s != null;
-		if (s.isEmpty())
+	public static NonNullPair<String, Boolean> getEnglishPlural(final String word) {
+		assert word != null;
+		if (word.isEmpty())
 			return new NonNullPair<>("", Boolean.FALSE);
-		for (final String[] p : plurals) {
-			if (s.endsWith(p[1]))
-				return new NonNullPair<>(s.substring(0, s.length() - p[1].length()) + p[0], Boolean.TRUE);
-			if (s.endsWith(p[1].toUpperCase(Locale.ENGLISH)))
-				return new NonNullPair<>(s.substring(0, s.length() - p[1].length()) + p[0].toUpperCase(Locale.ENGLISH), Boolean.TRUE);
+		for (final WordEnding ending : plurals) {
+			if (word.endsWith(ending.plural()))
+				return new NonNullPair<>(word.substring(0, word.length() - ending.plural().length()) + ending.singular(), Boolean.TRUE);
+			if (word.endsWith(ending.plural().toUpperCase(Locale.ENGLISH)))
+				return new NonNullPair<>(word.substring(0, word.length() - ending.plural().length()) + ending.singular().toUpperCase(Locale.ENGLISH), Boolean.TRUE);
 		}
-		return new NonNullPair<>(s, Boolean.FALSE);
+		return new NonNullPair<>(word, Boolean.FALSE);
 	}
 
 	/**
 	 * Gets the english plural of a word.
 	 *
-	 * @param s
+	 * @param word
 	 * @return The english plural of the given word
 	 */
-	public static String toEnglishPlural(final String s) {
-		assert s != null && s.length() != 0;
-		for (final String[] p : plurals) {
-			if (s.endsWith(p[0]))
-				return s.substring(0, s.length() - p[0].length()) + p[1];
+	public static String toEnglishPlural(final String word) {
+		assert word != null && word.length() != 0;
+		for (final WordEnding ending : plurals) {
+			if (word.endsWith(ending.singular()))
+				return word.substring(0, word.length() - ending.singular().length()) + ending.plural();
 		}
 		assert false;
-		return s + "s";
+		return word + "s";
 	}
 
 	/**
@@ -808,6 +803,25 @@ public abstract class Utils {
 				return false;
 		}
 		return true;
+	}
+
+	protected static class WordEnding { // To be a record in 2.10
+
+		private final String singular, plural;
+
+		private WordEnding(String singular, String plural) {
+			this.singular = singular;
+			this.plural = plural;
+		}
+
+		public String singular() {
+			return singular;
+		}
+
+		public String plural() {
+			return plural;
+		}
+
 	}
 
 }

--- a/src/main/java/ch/njol/skript/util/Utils.java
+++ b/src/main/java/ch/njol/skript/util/Utils.java
@@ -241,6 +241,7 @@ public abstract class Utils {
 
 	private final static String[][] plurals = {
 
+			{"ive", "ives"}, // objective fix
 			{"fe", "ves"},// most -f words' plurals can end in -fs as well as -ves
 
 			{"axe", "axes"},
@@ -259,7 +260,7 @@ public abstract class Utils {
 
 			{"man", "men"},
 
-		    {"ui", "uis"},
+		    {"ui", "uis"}, // gui fix
 			{"us", "i"},
 
 			{"hoe", "hoes"},

--- a/src/main/java/ch/njol/skript/util/Utils.java
+++ b/src/main/java/ch/njol/skript/util/Utils.java
@@ -66,15 +66,15 @@ import org.jetbrains.annotations.NotNull;
 
 /**
  * Utility class.
- * 
+ *
  * @author Peter Güttinger
  */
 public abstract class Utils {
-	
+
 	private Utils() {}
-	
+
 	public final static Random random = new Random();
-	
+
 	public static String join(final Object[] objects) {
 		assert objects != null;
 		final StringBuilder b = new StringBuilder();
@@ -85,7 +85,7 @@ public abstract class Utils {
 		}
 		return "" + b.toString();
 	}
-	
+
 	public static String join(final Iterable<?> objects) {
 		assert objects != null;
 		final StringBuilder b = new StringBuilder();
@@ -99,12 +99,12 @@ public abstract class Utils {
 		}
 		return "" + b.toString();
 	}
-	
+
 	@SuppressWarnings("unchecked")
 	public static <T> boolean isEither(@Nullable T compared, @Nullable T... types) {
 		return CollectionUtils.contains(types, compared);
 	}
-	
+
 	public static Pair<String, Integer> getAmount(String s) {
 		if (s.matches("\\d+ of .+")) {
 			return new Pair<>(s.split(" ", 3)[2], Utils.parseInt("" + s.split(" ", 2)[0]));
@@ -115,7 +115,7 @@ public abstract class Utils {
 		}
 		return new Pair<>(s, Integer.valueOf(-1));
 	}
-	
+
 //	public final static class AmountResponse {
 //		public final String s;
 //		public final int amount;
@@ -163,7 +163,7 @@ public abstract class Utils {
 
 	/**
 	 * Loads classes of the plugin by package. Useful for registering many syntax elements like Skript does it.
-	 * 
+	 *
 	 * @param basePackage The base package to add to all sub packages, e.g. <tt>"ch.njol.skript"</tt>.
 	 * @param subPackages Which subpackages of the base package should be loaded, e.g. <tt>"expressions", "conditions", "effects"</tt>. Subpackages of these packages will be loaded
 	 *            as well. Use an empty array to load all subpackages of the base package.
@@ -216,7 +216,7 @@ public abstract class Utils {
 
 	/**
 	 * The first invocation of this method uses reflection to invoke the protected method {@link JavaPlugin#getFile()} to get the plugin's jar file.
-	 * 
+	 *
 	 * @return The jar file of the plugin.
 	 */
 	@Nullable
@@ -240,12 +240,12 @@ public abstract class Utils {
 	}
 
 	private final static String[][] plurals = {
-			
+
 			{"fe", "ves"},// most -f words' plurals can end in -fs as well as -ves
-			
+
 			{"axe", "axes"},
 			{"x", "xes"},
-			
+
 			{"ay", "ays"},
 			{"ey", "eys"},
 			{"iy", "iys"},
@@ -254,28 +254,29 @@ public abstract class Utils {
 			{"kie", "kies"},
 			{"zombie", "zombies"},
 			{"y", "ies"},
-			
+
 			{"h", "hes"},
-			
+
 			{"man", "men"},
-			
+
+		    {"ui", "uis"},
 			{"us", "i"},
-			
+
 			{"hoe", "hoes"},
 			{"toe", "toes"},
 			{"o", "oes"},
-			
+
 			{"alias", "aliases"},
 			{"gas", "gases"},
-			
+
 			{"child", "children"},
-			
+
 			{"sheep", "sheep"},
-			
+
 			// general ending
 			{"", "s"},
 	};
-	
+
 	/**
 	 * @param s trimmed string
 	 * @return Pair of singular string + boolean whether it was plural
@@ -293,10 +294,10 @@ public abstract class Utils {
 		}
 		return new NonNullPair<>(s, Boolean.FALSE);
 	}
-	
+
 	/**
 	 * Gets the english plural of a word.
-	 * 
+	 *
 	 * @param s
 	 * @return The english plural of the given word
 	 */
@@ -309,10 +310,10 @@ public abstract class Utils {
 		assert false;
 		return s + "s";
 	}
-	
+
 	/**
 	 * Gets the plural of a word (or not if p is false)
-	 * 
+	 *
 	 * @param s
 	 * @param p
 	 * @return The english plural of the given word, or the word itself if p is false.
@@ -322,10 +323,10 @@ public abstract class Utils {
 			return toEnglishPlural(s);
 		return s;
 	}
-	
+
 	/**
 	 * Adds 'a' or 'an' to the given string, depending on the first character of the string.
-	 * 
+	 *
 	 * @param s The string to add the article to
 	 * @return The given string with an appended a/an and a space at the beginning
 	 * @see #A(String)
@@ -334,10 +335,10 @@ public abstract class Utils {
 	public static String a(final String s) {
 		return a(s, false);
 	}
-	
+
 	/**
 	 * Adds 'A' or 'An' to the given string, depending on the first character of the string.
-	 * 
+	 *
 	 * @param s The string to add the article to
 	 * @return The given string with an appended A/An and a space at the beginning
 	 * @see #a(String)
@@ -346,10 +347,10 @@ public abstract class Utils {
 	public static String A(final String s) {
 		return a(s, true);
 	}
-	
+
 	/**
 	 * Adds 'a' or 'an' to the given string, depending on the first character of the string.
-	 * 
+	 *
 	 * @param s The string to add the article to
 	 * @param capA Whether to use a capital a or not
 	 * @return The given string with an appended a/an (or A/An if capA is true) and a space at the beginning
@@ -367,14 +368,14 @@ public abstract class Utils {
 			return "a " + s;
 		}
 	}
-	
+
 	/**
 	 * Gets the collision height of solid or partially-solid blocks at the center of the block.
 	 * This is mostly for use in the {@link EffTeleport teleport effect}.
 	 * <p>
 	 * This version operates on numeric ids, thus only working on
 	 * Minecraft 1.12 or older.
-	 * 
+	 *
 	 * @param type
 	 * @return The block's height at the center
 	 */
@@ -530,14 +531,14 @@ public abstract class Utils {
 
 		return completableFuture;
 	}
-	
+
 	final static ChatColor[] styles = {ChatColor.BOLD, ChatColor.ITALIC, ChatColor.STRIKETHROUGH, ChatColor.UNDERLINE, ChatColor.MAGIC, ChatColor.RESET};
 	final static Map<String, String> chat = new HashMap<>();
 	final static Map<String, String> englishChat = new HashMap<>();
-	
+
 	public final static boolean HEX_SUPPORTED = Skript.isRunningMinecraft(1, 16);
 	public final static boolean COPY_SUPPORTED = Skript.isRunningMinecraft(1, 15);
-	
+
 	static {
 		Language.addListener(new LanguageChangeListener() {
 			@Override
@@ -554,21 +555,21 @@ public abstract class Utils {
 			}
 		});
 	}
-	
+
 	@Nullable
 	public static String getChatStyle(final String s) {
 		SkriptColor color = SkriptColor.fromName(s);
-		
+
 		if (color != null)
 			return color.getFormattedChat();
 		return chat.get(s);
 	}
-	
+
 	private final static Pattern stylePattern = Pattern.compile("<([^<>]+)>");
-	
+
 	/**
 	 * Replaces &lt;chat styles&gt; in the message
-	 * 
+	 *
 	 * @param message
 	 * @return message with localised chat styles converted to Minecraft's format
 	 */
@@ -602,11 +603,11 @@ public abstract class Utils {
 		m = ChatColor.translateAlternateColorCodes('&', "" + m);
 		return "" + m;
 	}
-	
+
 	/**
 	 * Replaces english &lt;chat styles&gt; in the message. This is used for messages in the language file as the language of colour codes is not well defined while the language is
 	 * changing, and for some hardcoded messages.
-	 * 
+	 *
 	 * @param message
 	 * @return message with english chat styles converted to Minecraft's format
 	 */
@@ -653,7 +654,7 @@ public abstract class Utils {
 	public static ChatColor parseHexColor(String hex) {
 		if (!HEX_SUPPORTED || !HEX_PATTERN.matcher(hex).matches()) // Proper hex code validation
 			return null;
-		
+
 		hex = hex.replace("#", "");
 		try {
 			return ChatColor.of('#' + hex.substring(0, 6));
@@ -661,10 +662,10 @@ public abstract class Utils {
 			return null;
 		}
 	}
-	
+
 	/**
 	 * Gets a random value between <tt>start</tt> (inclusive) and <tt>end</tt> (exclusive)
-	 * 
+	 *
 	 * @param start
 	 * @param end
 	 * @return <tt>start + random.nextInt(end - start)</tt>
@@ -733,12 +734,12 @@ public abstract class Utils {
 		// See #1747 to learn how it broke returning items from functions
 		return (Class<Found>) (chosen == Cloneable.class ? bestGuess : chosen == Object.class ? bestGuess : chosen);
 	}
-	
+
 	/**
 	 * Parses a number that was validated to be an integer but might still result in a {@link NumberFormatException} when parsed with {@link Integer#parseInt(String)} due to
 	 * overflow.
 	 * This method will return {@link Integer#MIN_VALUE} or {@link Integer#MAX_VALUE} respectively if that happens.
-	 * 
+	 *
 	 * @param s
 	 * @return The parsed integer, {@link Integer#MIN_VALUE} or {@link Integer#MAX_VALUE} respectively
 	 */
@@ -750,12 +751,12 @@ public abstract class Utils {
 			return s.startsWith("-") ? Integer.MIN_VALUE : Integer.MAX_VALUE;
 		}
 	}
-	
+
 	/**
 	 * Parses a number that was validated to be an integer but might still result in a {@link NumberFormatException} when parsed with {@link Long#parseLong(String)} due to
 	 * overflow.
 	 * This method will return {@link Long#MIN_VALUE} or {@link Long#MAX_VALUE} respectively if that happens.
-	 * 
+	 *
 	 * @param s
 	 * @return The parsed long, {@link Long#MIN_VALUE} or {@link Long#MAX_VALUE} respectively
 	 */
@@ -767,7 +768,7 @@ public abstract class Utils {
 			return s.startsWith("-") ? Long.MIN_VALUE : Long.MAX_VALUE;
 		}
 	}
-	
+
 	/**
 	 * Gets class for name. Throws RuntimeException instead of checked one.
 	 * Use this only when absolutely necessary.
@@ -783,7 +784,7 @@ public abstract class Utils {
 			throw new RuntimeException("Class not found!");
 		}
 	}
-	
+
 	/**
 	 * Finds the index of the last in a {@link List} that matches the given {@link Checker}.
 	 *
@@ -807,5 +808,5 @@ public abstract class Utils {
 		}
 		return true;
 	}
-	
+
 }

--- a/src/test/java/org/skriptlang/skript/test/tests/localization/UtilsPlurals.java
+++ b/src/test/java/org/skriptlang/skript/test/tests/localization/UtilsPlurals.java
@@ -51,9 +51,11 @@ public class UtilsPlurals {
 				{"kidney", "kidneys"},
 				{"anatomy", "anatomies"},
 				{"axe", "axes"},
-				{"elf", "elfs"},
 				{"knife", "knives"},
-				{"shelf", "shelfs"},
+				{"elf", "elves"},
+				{"shelf", "shelves"},
+				{"self", "selves"},
+				{"gui", "guis"},
 		};
 		for (String[] s : strings) {
 			assertEquals(s[1], Utils.toEnglishPlural(s[0]));

